model/job.go

```diff
@@ -17,36 +17,43 @@ import (
 	"gopkg.in/yaml.v2"
 )
 
-// JobLinkProvides describes the BOSH links a job provides
-type JobLinkProvides struct {
-	Name       string
-	Type       string
+// jobLinkInfo describes a BOSH link provider or consumer
+type jobLinkInfo struct {
+	Name     string `json:"-" yaml:"-"`
+	Type     string `json:"-" yaml:"-"`
+	RoleName string `json:"role" yaml:"-"`
+	JobName  string `json:"job" yaml:"-"`
+}
+
+// jobProvidesInfo describes a BOSH link provider
+type jobProvidesInfo struct {
+	jobLinkInfo
+	Alias      string `yaml:"as"`
+	Shared     bool   `yaml:"shared"`
 	Properties []string
 }
 
-// JobLinkConsumes describes the BOSH links a job consumes
-type JobLinkConsumes struct {
-	Name     string `json:"-"`
-	Type     string `json:"-"`
-	Optional bool   `json:"-"`
-	RoleName string `json:"role"` // Name of role in which the resolved job is running
-	JobName  string `json:"job"`  // Name of job that this resolves to
+// jobConsumesInfo describes the BOSH links a job consumes
+type jobConsumesInfo struct {
+	jobLinkInfo
+	Alias    string `yaml:"from"`
+	Optional bool
 }
 
 // Job represents a BOSH job
 type Job struct {
-	Name         string
-	Description  string
-	Templates    []*JobTemplate
-	Packages     Packages
-	Path         string
-	Fingerprint  string
-	SHA1         string
-	Properties   []*JobProperty
-	Version      string
-	Release      *Release
-	LinkProvides []*JobLinkProvides
-	LinkConsumes []*JobLinkConsumes
+	Name          string
+	Description   string
+	Templates     []*JobTemplate
+	Packages      Packages
+	Path          string
+	Fingerprint   string
+	SHA1          string
+	Properties    []*JobProperty
+	Version       string
+	Release       *Release
+	LinkProviders map[string]jobProvidesInfo
+	LinkConsumers []jobConsumesInfo
```

Comment:
Not sure why `LinkProviders` is a map and `LinkConsumers` just a plain array.

I guess there is currently no need to look up a consumer by name, but it would feel better to me if both were stored the same way for symmetry reasons.

Reply:
It would also more closely match the `RoleJob` struct in `model/roles.go`.

In that struct these 2 fields are called `Provides` and `Consumes` instead of `LinkProviders` and `LinkConsumers` (and both are maps). Maybe the naming should be unified?

Reply:
I'll work on unifying the name; the issue here is that consumer info might not have names (because they're matching by types), so there are no unique keys to use here. I'd have to switch to using lists for everything instead.

Reply:
Hmm. Since `RoleJob` embeds `Job`, it's actually accessible as `roleJob.LinkProviders` right now.  That's… going to be totally confusing.  Naming suggestions?

Reply:
I think using `Provides` and `Consumer` for both should be fine. When you are dealing with a `RoleJob` you really want the Union of both providers or consumers of the `RoleJob` and the referenced `Job`, right?

I kind of view the `Job` as the base class, with the `RoleJob` providing additional specialization.

Or maybe I'm already totally confused and am not making much sense. There is a reason I'm so slow in reviewing this PR...

Reply:
Renamed in 6a537f1